modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -258,16 +255,8 @@ const whitespaceCleaner = (root: AstNode, schema: Schema, settings: DomParserSet
   return [ preprocess, postprocess ];
 };
 
-const getRootBlockName = (settings: DomParserSettings, args: ParserArgs) => {
-  const name = args.forced_root_block ?? settings.forced_root_block;
-  if (name === false) {
-    return '';
-  } else if (name === true) {
-    return 'p';
-  } else {
-    return name;
-  }
-};
+const getRootBlockName = (settings: DomParserSettings, args: ParserArgs) =>
+  args.forced_root_block ?? settings.forced_root_block ?? '';
```

Comment:
I am not convinced adding `?? ''` is actually safe. Around line 487 where `getRootBlockName` is used, there is a check for if the block name is `undefined`. I don't know whether allowing the block name to be undefined is intentional or not but making this change seems risky

Reply:
I realised that where `getRootBlockName` is used around line 487, it is a falsy check so `''` will act the same as `undefined`. In any case, wherever we previously set `forced_root_block: false` in tests or other files, I would probably prefer for it to be changed to `forced_root_block: ''` rather than just removed.  Others might have different opinion.